File: dart-study/抽象类.dart
main(List<String> args) {
  //抽象类不能被实例化
  // var person = Person();//报错不能被实例化
}

abstract class Person{
    void info();
}

class Man extends Person{

  @override
  void info() {
    print("Man expand abstract Person");
  }

}

File: kotlin-study/抽象类abstract/抽象类.kt
//抽象类使用abstract关键字 抽象方法同样
abstract class Human(var name:String){
    abstract fun eat()
}

class Man(name:String):Human(name){
    override fun eat(){
        println("${name}比较能吃")
    }
}

class Woman(name:String):Human(name){
    override fun eat(){
        println("${name}不太能吃")
    }
}

fun main(args:Array<String>){
    var person1 = Man("林俊杰")
    person1.eat()
    var person2 = Woman("杨丞琳")
    person2.eat()
}